sdk/src/client.ts

```diff
@@ -1,79 +1,32 @@
-import { WebSocket } from 'ws';
-import type { AuthStorage } from './composables/authentication.js';
-import { withoutTrailingSlash } from './utils.js';
-// import type { ReadItemsInput, ReadItemsOutput } from './commands/read-items.js';
-// import type { Command } from './types/index.js';
-
 export interface ClientConfig {
-	fetch?: fetch;
-	ws?: typeof WebSocket;
-	url?: string;
 	token?: string;
 }
 
-export interface BaseClientConfig extends Record<string, any> {
-	fetch: fetch;
-	ws: typeof WebSocket;
-	apiURL: string;
-}
-
-export interface GenericClient<Schema extends object, Features extends object> {
-	config: BaseClientConfig;
-	auth: AuthStorage;
-
-	use: <ExtraFeatures extends object>(
-		feature: (client: DirectusClient<Schema, Features>) => ExtraFeatures
-	) => DirectusClient<Schema, Features & ExtraFeatures>;
+export interface DirectusClient<TSchema extends object> {
+	config: { url: URL };
+	auth: { token: string | null };
+	use: <TExtension extends object>(
+		createExtension: (client: DirectusClient<TSchema>) => TExtension
+	) => this & TExtension;
```

Comment:
By using `this & TExtension` typescript will correctly intersect on multiple composables used together